mixer/adapter/rbac/rbacStore.go

```diff
@@ -304,8 +424,52 @@ func valueMatch(a interface{}, b string) bool {
 	return false
 }
 
-// Check if all properties defined for the subject are satisfied by the properties from the request.
-func checkSubject(properties map[string]interface{}, subject map[string]string) bool {
+// checkSubjectsMatch returns true if rule subjects and instance subjects are matched. Otherwise
+// returns false.
+func checkSubjectsMatch(subject *rbacproto.Subject, instance *authorization.Instance) bool {
+	userMatch := false
+	groupsMatch := false
+	propertyMatch := false
+
+	if instance.Subject.User != "" {
+		if subject.GetUser() == "*" || subject.GetUser() == instance.Subject.User {
```

Comment:
I think the user could be prefix and suffix match. The same for groups.

Reply:
Used existing stringMatch function